### PR TITLE
KCRO-218: Failure to Save Message Metadata Due to Incorrect Context Status Check in cb_vmboxes Module

### PR DESCRIPTION
**Pull request by wdhammika**
Previously, it would fail when we tried to save voicemail media messages.
Now: Handle error causes that cause problems in saving records. (Remove extra validations)

### CODE REVIEW

File: src/modules/cb_vmboxes.erl [hunk 1]
```diff
@@ -601,19 +601,8 @@ check_mailbox_for_messages_array(Context, VMBoxMsgs) ->
...
-    case cb_context:resp_status(Context1) of
-        'success' ->
-            save_attachment(Context, cb_context:req_files(Context));
-        _ ->
-            Context1
-    end;
-maybe_save_attachment(Context, _Error) ->
-    lager:info("saving message metadata failed"),
-    Context.
+    save_attachment(Context1, cb_context:req_files(Context1)).
```
Comment (icehess) [inline src/modules/cb_vmboxes.erl:610]: lgtm, just please check resp_status in `put/3` before calling `maybe_save_attachment/1`

**Review by icehess — CHANGES_REQUESTED** [review]

File: src/modules/cb_vmboxes.erl (hunk 1 above)
Comment (wdhammika) [reply src/modules/cb_vmboxes.erl:610]: @icehess Could you please review the change?

**Review by icehess — APPROVED** [review]

**Comment by lazedo** [comment]
@wdhammika @icehess I can be wrong but this pr changes nothing

**Comment by icehess** [comment]
@lazedo there is also `put/5` calling `maybe_save_attachment/1` without any save operation and would fail because default context resp status is fatal.

**Comment by lazedo** [comment]
@icehess so the `fix` is wrong, it should instead set rest_status on validation

**Comment by icehess** [comment]
validation is fine, before execute request api_util sets resp status to fatal and all modules have to set it to success in their `put/post/delete/patch`. this module left this edge case and didn't set it.

**Comment by lazedo** [comment]
isn't that what I'm saying ? the proper fix would have been to set resp_status. this all thing of putting request api_util sets resp status to fatal still hitting us, not a good change.

**Pull request merged**